ping/ping_common.c

```diff
@@ -902,13 +902,17 @@ int finish(struct ping_rts *rts)
 		long tmavg = rts->tsum / total;
 		long long tmvar;
 
-		if (rts->tsum < INT_MAX)
-			/* This slightly clumsy computation order is important to avoid
-			 * integer rounding errors for small ping times. */
-			tmvar = (rts->tsum2 - ((rts->tsum * rts->tsum) / total)) / total;
-		else
-			tmvar = (rts->tsum2 / total) - (tmavg * tmavg);
-
+		// Add check for more than one to calculate std. dev. Will be zero otherwise.
+		if (total <= 1) {
+			tmvar = 0;
+		} else {
+			if (rts->tsum < INT_MAX)
+				/* This slightly clumsy computation order is important to avoid
+			 	* integer rounding errors for small ping times. */
+				tmvar = (rts->tsum2 - ((rts->tsum * rts->tsum) / total)) / (total - 1);
+			else
+				tmvar = (rts->tsum2 / (total - 1)) - (tmavg * tmavg) * (total / (total - 1));
+		}
```

Comment:
https://github.com/iputils/iputils/issues/332#issuecomment-848020207
> I'd be very hesitant to change the meaning of the existing fields, since they're well established at this point.  However, collecting and/or reporting additional statistics might be interesting.  It might be preferable to report the additional information only if explicitly requested at runtime (via a flag, etc), but I haven't thought too deeply about it.

I'd also be very careful to change this.


Just commenting style - reducing else part makes code more readable:
```
long long tmvar = 0;
...
if (total > 1) {
	if (rts->tsum < INT_MAX)
		/* This slightly clumsy computation order is important to avoid
		 * integer rounding errors for small ping times. */
		tmvar = (rts->tsum2 - ((rts->tsum * rts->tsum) / total)) / (total - 1);
	else
		tmvar = (rts->tsum2 / (total - 1)) - (tmavg * tmavg) * (total / (total - 1));
}
```
Also I'd just quash the commits into one. You also didn't put your `Signed-off-by: Your Name <your@email>`, see other commits.

Reply:
Thanks for you comments @pevik, this PR was from a long time ago before I knew what I know now, I honestly forgot about it. I presume since I deleted my fork, it notified you. Thank you for your comments however!